theano/sandbox/cuda/tests/test_cula.py

```diff
@@ -67,3 +67,111 @@ def test_uni_rand_solve(self):
         x_val = numpy.random.uniform(-0.4, 0.4,
                                      (A_val.shape[1], 4)).astype("float32")
         self.run_gpu_solve(A_val, x_val)
+
+
+class TestGpuCholesky(unittest.TestCase):
+
+    def setUp(self):
+        utt.seed_rng()
+
+    def get_gpu_cholesky_func(self, lower=True, inplace=False):
+        """ Helper function to compile function from GPU Cholesky op. """
+        A = theano.tensor.matrix("A", dtype="float32")
+        cholesky_op = cula.GpuCholesky(lower=lower, inplace=inplace)
+        chol_A = cholesky_op(A)
+        return theano.function([A], chol_A, accept_inplace=inplace)
+
+    def compare_gpu_cholesky_to_numpy(self, A_val, lower=True, inplace=False):
+        """ Helper function to compare op output to numpy.cholesky output. """
+        chol_A_val = numpy.linalg.cholesky(A_val)
+        if not lower:
+            chol_A_val = chol_A_val.T
+        fn = self.get_gpu_cholesky_func(lower, inplace)
+        res = fn(A_val)
+        chol_A_res = numpy.array(res)
+        utt.assert_allclose(chol_A_res, chol_A_val)
+
+    def test_invalid_input_fail_non_square(self):
+        """ Invalid Cholesky input test with non-square matrix as input. """
+        def invalid_input_func():
+            A_val = numpy.random.normal(size=(3, 2)).astype("float32")
+            fn = self.get_gpu_cholesky_func(True, False)
+            fn(A_val)
+        self.assertRaises(ValueError, invalid_input_func)
```

Comment:
This can be simplified. You don't need to make a new method. So just do something like:

def test_...(self):
    A_val = ...
   fn = ...
   self.assertRaises(ValueError, fn, A_val).

idem for the other tests.